src/chains/ethereum/transaction/tests/index.test.ts

```diff
@@ -1,6 +1,317 @@
 import assert from "assert";
-import ethereumTransaction from "../";
+import {
+  EIP2930AccessListTransaction,
+  LegacyTransaction,
+  TransactionFactory,
+  TypedDatabaseTransaction,
+  TypedRpcTransaction
+} from "../../transaction";
+import Common from "@ethereumjs/common";
+import Wallet from "../../ethereum/src/wallet";
+import { EthereumOptionsConfig } from "../../options";
+import { BUFFER_EMPTY, Quantity } from "@ganache/utils";
 
-describe("@ganache/ethereum-transaction", () => {
-  it("needs tests");
+describe("@ganache/ethereum-transaction", async () => {
+  const common = Common.forCustomChain(
+    "mainnet",
+    {
+      name: "ganache",
+      chainId: 1337,
+      comment: "Local test network",
+      bootstrapNodes: []
+    },
+    "berlin"
+  );
+  const options = EthereumOptionsConfig.normalize({});
+  const wallet = new Wallet(options.wallet);
+  const [from, to, accessListAcc] = wallet.addresses;
+  const fromBuf = Quantity.from(from).toBuffer();
+  const fakePrivateKey = Buffer.concat([fromBuf, fromBuf.slice(0, 12)]);
+
+  describe("TransactionFactory", () => {
+    describe("LegacyTransaction type from factory", () => {
+      let txFromRpc: LegacyTransaction;
+      it("generates legacy transactions from rpc data", async () => {
+        const rpc: TypedRpcTransaction = {
+          from: from,
+          to: to,
+          type: "0x0",
+          gasPrice: "0xffff"
+        };
+        txFromRpc = <LegacyTransaction>TransactionFactory.fromRpc(rpc, common);
+        txFromRpc.signAndHash(fakePrivateKey);
```

Comment:
do we need to `signAndHash` here?

Reply:
Signing and hashing fills out the v/r/s so that we get a legitimate raw transaction in the next two tests. That being said, this can definitely be cleaner. If we can make the accounts that we use the same for every run of the tests, couldn't we just hard-code the tx data to test?

Reply:
Since I've updated to use deterministic accounts, we can also have hard-coded tx data, so this is no longer needed.